cmake/FindFilesystem.cmake

```diff
@@ -0,0 +1,247 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
+# file Copyright.txt or https://cmake.org/licensing for details.
+
+#[=======================================================================[.rst:
+
```

Comment:
could we just add a brief comment with a link to where this was taken from? Just so if anything breaks one can check upstream for a potential fix?

Reply:
The thing is that I do not know the original author 😅 

I took this one from the link I shared previously from the CMake Issue:
https://gitlab.kitware.com/cmake/cmake/-/issues/17834

In there, somebody comments about this file existing at least in 2 repositories:
https://github.com/lethal-guitar/RigelEngine/blob/master/cmake/Modules/FindFilesystem.cmake
https://github.com/vector-of-bool/CMakeCM/blob/master/modules/FindFilesystem.cmake

Would it be enough to just have it here (in the PR discussion)? If something ever happen here, git blame should help to obtain more information.

Reply:
Since this file shouldn't really change, I guess that's fine :+1: